lib/plugins/rotate_asg_instances.rb

```diff
@@ -7,11 +7,41 @@ module Moonshot
   module Plugins
     # Rotate ASG instances after update.
     class RotateAsgInstances
-      def post_update(resources)
-        asg = ASG.new(resources)
+      include ::Moonshot::DoctorHelper
+
+      def initialize(ssh_user = ENV['MOONSHOT_SSH_USER'])
+        @ssh_user = ssh_user
+      end
+
+      def doctor(resources)
+        @resources = resources
+        doctor_check_ssh
+      end
+
+      def pre_update(resources)
+        @resources = resources
+        asg.try_to_ssh(raise_exception: true)
+      end
+
+      def post_update(_resources)
         asg.rotate_asg_instances
         asg.teardown_outdated_instances
       end
+
+      private
+
+      def doctor_check_ssh
+        if asg.try_to_ssh
+          success('Successfully opened SSH connection to a worker instance.')
+        else
+          critical('SSH connection test failed, check your SSH settings' \
```

Comment:
Here we are hiding the actual error, should the asg.try_to_ssh raise an exception and we rescue here and add the exception message to the critical log?

Reply:
In case of doctor check, we do not raise an exception in the `execute_ssh` and instead we receive `true/false` from the method and raise a custom exception in this block.